ambry-frontend/src/main/java/com/github/ambry/frontend/s3/S3ListHandler.java

```diff
@@ -99,14 +100,16 @@ private ReadableStreamChannel serializeAsXml(RestRequest restRequest, Page<Named
     String delimiter = getHeader(restRequest.getArgs(), DELIMITER_PARAM_NAME, false);
     String encodingType = getHeader(restRequest.getArgs(), ENCODING_TYPE_PARAM_NAME, false);
     String maxKeys = getHeader(restRequest.getArgs(), MAXKEYS_PARAM_NAME, false);
-    int maxKeysValue = maxKeys == null ? Integer.MAX_VALUE : Integer.parseInt(maxKeys);
+    //By default S3 list returns up to 1000 key names.
+    int maxKeysValue = maxKeys == null ? DEFAULT_MAX_KEY_VALUE : Integer.parseInt(maxKeys);
```

Comment:
Do we wanna change the list interface from NamedBlobDB to allow passing a maxKeyCount? In this way, we can control how many results to query in database.

Reply:
I can create a follow up pr to support this.